### PR TITLE
feat(app): add TxDetails component

### LINKED ISSUE #35: Create TxDetails component
## Figma Reference
https://www.figma.com/file/uEckQY4EaQ5rXwyISg6d1g/Fuel-Wallet?node-id=84%3A6962

## Props
- `transaction: Transaction`

## Subcomponents
- `<TxDetails.Loading>`
  - This component is just a loading state using our Skeleton

### PR DESCRIPTION
**Pull request by pedronauck**
✨ Fix #35

### CODE REVIEW

File: packages/app/src/systems/Transaction/utils/gas.test.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import { bn } from 'fuels';
+
+import { getGasUsedFromTx } from './gas';
+
+const TX = {
+  receipts: [
+    { gasUsed: bn(10000000) },
```
Comment (luizstacio) [inline packages/app/src/systems/Transaction/utils/gas.test.ts:8]: We only have one receipt with `gasUsed`, it has the receipt type `ReceiptType.ScriptResult`

File: packages/app/src/systems/Transaction/utils/gas.tsx [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import type { ReceiptScriptResult } from 'fuels';
+import { bn } from 'fuels';
+import type { Maybe } from 'yup/lib/types';
+
+import type { Transaction } from '../types';
+
+export function getGasUsedFromTx(tx?: Maybe<Transaction>) {
```
Comment (luizstacio) [inline packages/app/src/systems/Transaction/utils/gas.tsx:7]: By having just a single receipt you should be able to remove this.

**Review by luizstacio — CHANGES_REQUESTED** [review]

File: packages/app/src/systems/Transaction/utils/gas.test.ts (hunk 1 above)
Comment (pedronauck) [reply packages/app/src/systems/Transaction/utils/gas.test.ts:8]: ✅

File: packages/app/src/systems/Transaction/components/TxDetails/TxDetails.tsx [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+import type { Maybe } from '~/systems/Core';
+import { formatUnits } from '~/systems/Core';
+
+export type TxDetailsProps = {
+  tx?: Maybe<Transaction>;
+};
+
+export function TxDetails({ tx }: TxDetailsProps) {
+  const gasUsed = useMemo(() => {
```
[the remark this answers is not in the document]
Comment (pedronauck) [reply packages/app/src/systems/Transaction/components/TxDetails/TxDetails.tsx]: cool, done

**Review by luizstacio — APPROVED** [review]

**Pull request merged**